File: Week2/2_last_digit_of_fibonacci_number/fibonacci_last_digit.cpp
#include <iostream>

using namespace std;

long fibonacci_last_digit(long long n){

    int first = 0;
    int second = 1;
    int i;
    int res;

    for (i = 2; i <=n ; i++) {
        res = (first + second) % 10;
        first = second;
        second =res;

    }
    return res;
}

int main(){
    int n;
    cout <<"Enter the nth number you want to calculate for Fibonacci ";
    cin >> n;

   cout << fibonacci_last_digit(n);

}

File: Week2/2_fibonacci_last_digit/fibonacci_last_digit.py
def fibonacci_last_digit(n):
    first = 0
    second = 1

    for i in range(2, n+1, 1):
        res = (first + second) % 10
        first = second
        second = res

    return res

if __name__ == '__main__':

    n = input("Enter the nth number you want to caculate for fibonacci: ")
    print(fibonacci_last_digit(int(n)))
